srcStatic/FileSystemHelper.cpp

```diff
@@ -40,3 +41,29 @@ bool Exists(const std::string& path)
 	auto modifiedPath = fs::path(path) / ".";
 	return fs::exists(modifiedPath);
 }
+
+std::vector<std::string> LocateFilesWithExtension(const std::string& directory, std::string extension)
+{
+	if (!IsDirectory(directory)) {
+		return {};
+	}
+
+	if (extension.empty()) {
+		throw std::runtime_error("provided extension cannot be empty");
+	}
```

Comment:
I'm fine with allowing empty extensions. For this function to be proper, I believe each directory entry needs to be checked to make sure it is a file. Otherwise an empty extension might return all the subdirectories. Also, a directory named bad.vol might pass right now.

Since this function is changed by another branch, I'm going to hold off working on it for now.

Reply:
Empty extensions doesn't have anything to do with file versus directory. They are orthogonal concerns.

You do raise a valid point about not wanting to pass a directory name to be loaded as a VOL file.